d3-scale-chromatic - Adding missing schemes

henriquefm opened the pull request:
Library documentation:
https://github.com/d3/d3-scale-chromatic

Commit that added the missing schemes:
https://github.com/d3/d3-scale-chromatic/commit/ade54c13e8dfdb9807801a794eaec1a37f926b8a

Adding the schemes:
// Diverging
schemeBrBG
schemePRGn
schemePiYG
schemePuOr
schemeRdBu
schemeRdGy
schemeRdYlBu
schemeRdYlGn
schemeSpectral
// Sequential
schemeBlues
schemeGreens
schemeGreys
schemeOranges
schemePurples
schemeReds
// Sequential(Multi-Hue)
schemeBuGn
schemeBuPu
schemeGnBu
schemeOrRd
schemePuBuGn
schemePuBu
schemePuRd
schemeRdPu
schemeYlGnBu
schemeYlGn
schemeYlOrBr
schemeYlOrRd

gustavderdrache commented on types/d3-scale-chromatic/index.d.ts:
```diff
@@ -1,6 +1,9 @@
 // Type definitions for D3JS d3-scale-chromatic module 1.0
```
Since d3/d3-scale-chromatic@ade54c1 is part of v1.1.1, let's update this header to say 1.1 instead of 1.0.

gustavderdrache commented on types/d3-scale-chromatic/index.d.ts:
```diff
@@ -45,173 +48,269 @@ export const schemeSet3: string[];
+ *  (often used with d3.scaleOrdinal). Each discrete scheme, such as d3.schemeBrBG, is represented as an array of arrays of hexadecimal
+ *  color strings. The kth element of this array contains the color scheme of size k; for example, d3.schemeBrBG[9] contains an array
+ *  of nine strings representing the nine colors of the brown-blue-green diverging color scheme. Diverging color schemes support
+ *  a size k ranging from 3 to 11.
```
In editors that support autocompletion, this information is attached only to the `interpolateBrBG` function. I suggest removing it, save for the information about the _k_ variable, which should get repeated for all of the `const scheme*` variables.

gustavderdrache commented on types/d3-scale-chromatic/index.d.ts:
```diff
@@ -45,173 +48,269 @@ export const schemeSet3: string[];
  */
 export function interpolateBrBG(value: number): string;
+
+export const schemeBrBG: string[][];
```
We've had some issues with arrays (see, for example, the discussion in #18580). I think these should be `ReadonlyArray<ReadonlyArray<string>>` instead of `string[][]` to both prevent modification and relax the usage requirements.

gustavderdrache commented on types/d3-scale-chromatic/index.d.ts:
```diff
@@ -45,173 +48,269 @@ export const schemeSet3: string[];
+ *  as discrete schemes (often used with d3.scaleOrdinal). Each discrete scheme, such as d3.schemeBlues, is represented as
+ *  an array of arrays of hexadecimal color strings. The kth element of this array contains the color scheme of size k; for example,
+ *  d3.schemeBlues[9] contains an array of nine strings representing the nine colors of the blue sequential color scheme.
+ *  Sequential, single-hue color schemes support a size k ranging from 3 to 9.
```
Same as the above comment regarding the divergent scales: this comment only applies to the `interpolateBlues` function.

henriquefm commented:
Thanks for the comments, I appreciate them.

> Since d3/d3-scale-chromatic@ade54c1 is part of v1.1.1, let's update this header to say 1.1 instead of 1.0.

Ok, changed.

> In editors that support autocompletion, this information is attached only to the interpolateBrBG function. I suggest removing it, save for the information about the k variable, which should get repeated for all of the const scheme* variables.

I just reproduced the info from https://github.com/d3/d3-scale-chromatic, with the same order and everything, didn't think about how it would affect the autocompletion. Now it's better.

> We've had some issues with arrays (see, for example, the discussion in #18580). I think these should be ReadonlyArray<ReadonlyArray<string>> instead of string[][] to both prevent modification and relax the usage requirements.

Made these changes. Also changed the existing schemes:
`export const schemeAccent: string[];`
to
`export const schemeAccent: ReadonlyArray<string>;`

Thanks!
